compiler/ballerina-lang/src/main/java/io/ballerina/projects/DocumentContext.java

```diff
@@ -183,7 +183,7 @@ private ModuleLoadRequest getModuleLoadRequest(ImportDeclarationNode importDcl,
 
         // Create the module load request
         return new ModuleLoadRequest(orgName, packageName, moduleName, null, scope,
-                                     DependencyResolutionType.SOURCE);
+                                     DependencyResolutionType.SOURCE, importDcl.location());
```

Comment:
Adding the location to the `ModuleLoadRequest` does not work. `ModueContext` class stores `ModuleLoadRequest`s in a `Set` to remove duplicate. If we create two `ModuleLoadRequest` instances with the same org, package with different locations, only the first one is stored in the `ModuleContext`.

Reply:
Removed